clients/client/src/main/java/org/projectnessie/client/StreamingUtil.java

```diff
@@ -93,29 +109,28 @@ private StreamingUtil() {
    * limited, returns all commit-log entries.
    *
    * @param api The {@link NessieApiV1} to use
-   * @param ref a named reference (branch or tag name) or a commit-hash
+   * @param ref a named reference (branch or tag name).
+   * @param builderCustomizer a Function that takes and returns {@link GetCommitLogBuilder} that is
+   *     passed to the caller of this method. It allows the caller to customize the
+   *     GetCommitLogBuilder with additional parameters (e.g.: hashOnRef, untilHash, filter,
+   *     fetchOption).
+   * @param maxRecords - a maximum number of records in the stream. If it is {@code
+   *     Optional.empty()} the stream will be unbounded.
    * @return stream of {@link CommitMeta} objects
    */
   public static Stream<LogEntry> getCommitLogStream(
       @NotNull NessieApiV1 api,
-      @NotNull String ref,
-      @Nullable String hashOnRef,
-      @Nullable String untilHash,
-      @Nullable String filter,
-      OptionalInt maxRecords,
-      boolean fetchAdditionalInfo)
+      Reference ref,
+      @NotNull Function<GetCommitLogBuilder, GetCommitLogBuilder> builderCustomizer,
+      @NotNull OptionalInt maxRecords)
       throws NessieNotFoundException {
-    FetchOption fetchOption = fetchAdditionalInfo ? FetchOption.ALL : FetchOption.MINIMAL;
     return new ResultStreamPaginator<>(
             LogResponse::getLogEntries,
-            (reference, pageSize, token) ->
-                builderWithPaging(api.getCommitLog().fetch(fetchOption), pageSize, token)
-                    .refName(reference)
-                    .hashOnRef(hashOnRef)
-                    .filter(filter)
-                    .untilHash(untilHash)
-                    .get())
-        .generateStream(ref, maxRecords);
+            (reference, pageSize, token) -> {
+              GetCommitLogBuilder builder = builderCustomizer.apply(api.getCommitLog());
+              return builderWithPaging(builder, pageSize, token).refName(reference).get();
+            })
+        .generateStream(ref.getHash(), maxRecords);
```

Comment:
The last question: 
I see that `ref` that is passed here may be `Detached.of(hash)`, but also `Reference.of(name,null)`. For that reason, is it safe to pass hash here or should we have a conditional logic:
`ref.getHash != null ? ref.getHash() : ref.getName()`
?